### DISCUSSION TITLE
Change the wallet connect user flow to be more intuitive and to work with a greater variety of wallets

### DISCUSSION THREAD

**Discussion by polkadirk** [OAK-Foundation/OAK-blockchain#489] [Automation Hub]
Currently when connecting a wallet the user is prompted to select a wallet extension from metamask or polkadot js, example below. This should instead prompt the user to select an EVM or Substate wallet first, and then have the option to connect metamask or talisman wallet for EVM and pjs, talisman wallet and subwallet for substrate (nova wallet if we want mobile support for the page).

This user flow is more inline with other dapps that use both substrate and evm wallet extensions and especially for users with Talisman as they can use the one extension for both ecos.

![111](https://github.com/OAK-Foundation/OAK-blockchain/assets/104081857/580477d4-394b-466d-b28e-1fd6cf527525)

**Comment by chrisli30** [comment]
Yes, I agree. We’ll add the support to Talisman and Subwallet in a later version. 👌
